qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -813,8 +813,20 @@ def _run_analysis(
                 ) from ex
 
             meas_return = run_options.get("meas_return", None)
-
-            data_processor = get_processor(meas_level, meas_return, self.options.normalization)
+            init_qubits = run_options.get("init_qubits")
+            memory = run_options.get("memory")
+            rep_delay = run_options.get("rep_delay")
+            num_qubits = self._num_qubits
+
+            data_processor = get_processor(
+                meas_level,
+                meas_return,
+                self.options.normalization,
+                init_qubits,
+                memory,
+                rep_delay,
+                num_qubits,
+            )
```

Comment:
Can you try to implement something like this?
```suggestion
            data_processor = get_processor(**run_options, **self._options, num_qubits)
```